### PR TITLE
Zib-FunctionalOrMentalStatus

### PR DESCRIPTION
**Pull request by Ashwin-nictiz**
Hi @ArdonToonstra and @jduwel ,

Can you review this for me?

Double check list:

- Comment about MedicalDevice reference in root element
- ValueSetsBinding 

Issues:

> +++ Validating other terminology
> == ./resources/zib/terminology/StatusNaamICFCodelijst-2.16.840.1.113883.2.4.3.11.60.40.2.4.26.3--20200901000000.xml (2.16.840.1.113883.2.4.3.11.60.40.2.4.26.3--20200901000000)
>   -  error at ValueSet.compose.include[0].system (38, 61):
>      URL value 'http://hl7.org/fhir/sid/icf-nl' does not resolve

> == ./resources/zib/terminology/StatusWaardeICFCodelijst-2.16.840.1.113883.2.4.3.11.60.40.2.4.26.6--20200901000000.xml (2.16.840.1.113883.2.4.3.11.60.40.2.4.26.6--20200901000000)
>   -  error at ValueSet.compose.include[0].system (38, 61):
>      URL value 'http://hl7.org/fhir/sid/icf-nl' does not resolve

> ==== snapshots/zib-FunctionalOrMentalStatus.json
>      == NL-CM:4.26.3 (Observation.value[x]:valueCodeableConcept)
>         datatype:    ERROR (CodeableConcept instead of CO)

Comment from Alexander about the deprecated ValueSets:

> Die verouderde waardelijsten moeten gewoon mee. Ze worden ontraden om nog te gebruiken voor nieuwe registraties maar bestaande registraties veranderen daar niet door.

### CODE REVIEW

File: release-notes.md [hunk 1]
```diff
@@ -27,6 +27,16 @@ This document contains release notes per zib, indicating differences with their
...
 
+## zib-FunctionalOrMentalStatus
+* SNOMED codes functionalCode and mentalCode on `Observation.category` have been removed
+* The value lists StatusNameICFCodelist, StatusNameNOCCodelist, StatusNameOMAHACodelist, StatusNameSnomedCodelist are combined into one list and modulated on `Observation.code`
+* The value lists StatusValueICFCodelist, StatusValueNOCCodelist, StatusValueOMAHACodelist, StatusValueSnomedCodelist are combined into on list and modulated on `Observation.value[x]:valueCodeableConcept`
+* `EffectiveDateTime` is modulated on its own slice
+* The comment element is modulated on `Observation.note.text` instead of `Observation.comment`
+* MedicalDevice extension have been removed see comment on root element for the explanation
+* The patient reference has been moved to the nl-core layer
+
```
Comment (jduwel) [inline release-notes.md:38]: Ik vind de release notes wel heel erg gedetailleerd. Kan hoor (wie ben ik), maar volgens mij hoeven hier alleen de wijzigingen die erg opvallend zijn of die niet helemaal volgens de guidelines zijn. Alles wat 'logisch' is, hoeft hier niet volgens mij.

Also: je gebruikt vaak 'modulated' - maar dat klopt niet. Bedoel je 'modelled'? Of 'mapped'?

File: resources/zib/zib-FunctionalOrMentalStatus.xml [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+  <kind value="resource" />
+  <abstract value="true" />
+  <type value="Observation" />
+  <baseDefinition value="http://hl7.org/fhir/StructureDefinition/Observation" />
+  <derivation value="constraint" />
+  <differential>
+    <element id="Observation">
+      <path value="Observation" />
+      <short value="FunctionalOrMentalStatus" />
+      <comment value="The data element MedicalDevice with id NL-CM:4.26.5 is not modulated in this profile because, the reference is modulated the other way around. The resource DeviceUseStatement has an FHIR element reasonReference that has a reference to an observation resource. Because of this reference, it is not necessary to include a MedicalDevice extension in this profile." />
```
Comment (jduwel) [inline resources/zib/zib-FunctionalOrMentalStatus.xml:35]: 'Modulated' klopt niet in deze zin. Bovendien is dit een erg lange zin om te zeggen wat je wil zeggen. Volgens mij kan het compacter:

> The reference to zib-MedicalDevice in concept 'MedicalDevice' (NL-CM:4.26.5), is reversed in FHIR. The relation is mapped in zib-MedicalDevice on `DeviceUseStatement.reasonReference`.

Heb je ergens bij zib-MedicalDevice (Ardon doet die) duidelijk gemaakt dat deze mapping in dat profiel moet komen? Of je moet hem zelf alvast toevoegen

**Review by jduwel — CHANGES_REQUESTED** [review]
known-issue:

==== snapshots/zib-FunctionalOrMentalStatus.json
     == NL-CM:4.26.3 (Observation.value[x]:valueCodeableConcept)
        datatype:    ERROR (CodeableConcept instead of CO)

Nog oplossen

File: resources/zib/zib-FunctionalOrMentalStatus.xml (hunk 2 above)
Comment (Ashwin-nictiz) [reply resources/zib/zib-FunctionalOrMentalStatus.xml:35]: Fixt

**Comment by ArdonToonstra** [comment]
During the mapping phase problems arose from providing a value for the mandatory `.category` element. Created a zib ticket: https://bits.nictiz.nl/browse/ZIB-1549

File: known-issues.yml [hunk 3]
```diff
@@ -30,6 +30,15 @@ pattern-ZibHealthProfessionalReference:
 
 zib-AddressInformation:
   zib deviations:
+    Address.line.extension:streetName.value[x]:
```
Comment (LuudSlagter) [inline known-issues.yml:33]: There are some additions that do not seem to be related to this branch (I don't know if this is a problem).
Note: the changes w.r.t. EpisodeOfCare/HeadCircumference/HeartRate are the consequence of changing these to the right alphabetical order.

File: resources/zib/zib-FunctionalOrMentalStatus.xml [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+          <path value="$this" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+      <min value="1" />
+      <constraint>
+        <key value="zib-FunctionalOrMentalStatus-1" />
+        <severity value="warning" />
+        <human value="Either a category for funtional status finding or mental status finding is present." />
+        <expression value="where(coding.system = 'http://snomed.info/sct' and coding.code = '118228005').exists() or where(coding.system = 'http://snomed.info/sct' and coding.code = '384821006').exists()" />
```
Comment (LuudSlagter) [inline resources/zib/zib-FunctionalOrMentalStatus.xml:57]: 'Either' in line 56 suggests that only one option is possible (i.e. a status finding cannot be both functional and mental), which makes sense. Should we then change the 'or' to 'xor' (or the logical equivalent of ' xor'), or is that too restricting?
If however it is possible for a finding to belong to both categories, maybe we should change 'Either a' to 'At least one' or something similar.

File: resources/zib/zib-FunctionalOrMentalStatus.xml [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+          <path value="$this" />
+        </discriminator>
+        <rules value="open" />
+      </slicing>
+    </element>
+    <element id="Observation.value[x]:valueCodeableConcept">
+      <path value="Observation.value[x]" />
+      <sliceName value="valueCodeableConcept" />
+      <short value="StatusValue" />
+      <definition value="The extent to which the patient is limited by the functional or mental condition. &#xD;&#xA;If both the status name and the status value are recorded in code, then both values must come from the same code system.  &#xD;&#xA;If the patient uses a medical device, the evaluation of the extent of the limitation will take place without this aid.&#xD;&#xA;The answer codes must be selected from the subselection of the code system corresponding with the request in accordance with the rules applicable to that system." />
```
Comment (LuudSlagter) [inline resources/zib/zib-FunctionalOrMentalStatus.xml:137]: Should we include this as a constraint or is that too restricting?
Something like: if .code.coding.system and .valueCodeableConcept.coding.system are not empty, then .code.coding.system = .valueCodeableConcept.coding.system.

File: examples/nl-core-FunctionalOrMentalStatus-01.xml [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+   <effectiveDateTime value="2015-02-23"/>
+   <valueCodeableConcept>
+      <coding>
+         <system value="http://snomed.info/sct"/>
+         <code value="284892005"/>
+         <display value="moeite met scheren"/>
+      </coding>
+   </valueCodeableConcept>
+   <note>
+      <text value="Dit is een toelichting"/>
```
Comment (LuudSlagter) [inline examples/nl-core-FunctionalOrMentalStatus-01.xml:31]: Could we include a more speaking comment, such as 'Door verminderde fijne motoriek niet mogelijk om het gezicht volledig glad te scheren'

**Review by LuudSlagter — CHANGES_REQUESTED** [review]

File: known-issues.yml (hunk 3 above)
Comment (ArdonToonstra) [reply known-issues.yml:33]: Likely due to merging main into this branch and not resolving the conflicts correctly... Or not sure actually. Strange that this is a change compared to the main branch.
@pieter-edelman-nictiz , do you think this is incorrect?

File: examples/nl-core-FunctionalOrMentalStatus-01.xml (hunk 6 above)
Comment (ArdonToonstra) [reply examples/nl-core-FunctionalOrMentalStatus-01.xml:36]: Will do! Thanks.

File: resources/zib/zib-FunctionalOrMentalStatus.xml (hunk 4 above)
Comment (ArdonToonstra) [reply resources/zib/zib-FunctionalOrMentalStatus.xml:57]: Good suggestion! Thnx. 
I changed the operator to xor. The `.category` element s 1..*, with the intention to use one of these codes, so I would say it is not too restricting.

File: resources/zib/zib-FunctionalOrMentalStatus.xml (hunk 5 above)
Comment (ArdonToonstra) [reply resources/zib/zib-FunctionalOrMentalStatus.xml:137]: A good suggestion! I would say it is not too restricting but it will require a complex invariant. I am not sure if that would outweigh the benefits. 

It will be complex because `.coding` is repeating and often multiple codings and systems are given in random order... you can only test for sure that both lists the `.code.coding.system` and the `.valueCodeableConcept.coding.system` need to have at least one system that is present in both lists. 

This was my first thoughts:
> (code.coding.system.exists() and valueCodeableConcept.coding.system.exists())  implies code.coding.where(system.subsetOf(valueCodeableConcept.coding.system)).exists()

But yeah again, I think we should skip this one. Unless you can shake something from you sleeve :p

File: resources/zib/zib-FunctionalOrMentalStatus.xml (hunk 4 above)
Comment (LuudSlagter) [reply resources/zib/zib-FunctionalOrMentalStatus.xml:57]: This change has not yet been committed, right? (You're probably waiting on the possible other changes)

File: resources/zib/zib-FunctionalOrMentalStatus.xml (hunk 5 above)
Comment (LuudSlagter) [reply resources/zib/zib-FunctionalOrMentalStatus.xml:137]: We could use the following expression:
`(code.coding.system.exists() and valueCodeableConcept.coding.system.exists()) implies (code.coding.system.distinct().subsetOf(valueCodeableConcept.coding.system.distinct()) and code.coding.system.distinct().supersetOf(valueCodeableConcept.coding.system.distinct()))`
By checking on both `subsetOf()` and `supersetOf()` we're actually checking equality of both sets. I've added `distinct()` because one might add multiple codes of the same code system. Do you think this constraint adds enough value to warrant such an expression? ;)

File: known-issues.yml (hunk 3 above)
Comment (pieter-edelman-nictiz) [reply known-issues.yml:33]: Looks like a failed merge conflict, I'll see if I can fix it. That being said, please don't touch the order in a branch; it makes merge conflicts so much harder to resolve. If there's something wrong, it should be fixed upstream.

**Comment by pieter-edelman-nictiz** [comment]
Note: this PR is blocked by #140. When this issue is resolved, it should be fixed in known-issues.yml.

File: resources/zib/zib-FunctionalOrMentalStatus.xml (hunk 5 above)
Comment (pieter-edelman-nictiz) [reply resources/zib/zib-FunctionalOrMentalStatus.xml:137]: Let's say we have for `.code` a coding in SNOMED and an additional coding in LOINC and for `.valueCodeableConcept` a coding in SNOMED. Then `.coding` is not a subset of `.valueCodeableConcept`. I think this is allowed though.

**Comment by pieter-edelman-nictiz** [comment]
Manually merged, closing this PR
